third-party/crowdsec/src/main/java/org/apache/james/model/CrowdsecHttpClient.java

```diff
@@ -0,0 +1,81 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.model;
+
+import static org.apache.james.model.CrowdsecClientConfiguration.DEFAULT_API_KEY;
+import static org.apache.james.model.CrowdsecClientConfiguration.DEFAULT_TIMEOUT_IN_SECONDS;
+
+import java.time.Duration;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
+
+import reactor.core.publisher.Mono;
+import reactor.netty.http.client.HttpClient;
+
+public class CrowdsecHttpClient {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CrowdsecHttpClient.class);
+    public static final String GET_DECISION = "/decisions";
+
+    private final HttpClient httpClient;
+    private final ObjectMapper mapper;
+
+    @Inject
+    public CrowdsecHttpClient(CrowdsecClientConfiguration configuration) {
+        this.httpClient = buildReactorNettyHttpClient(configuration);
+        this.mapper = new ObjectMapper().registerModule(new Jdk8Module());
+    }
+
+    public Mono<List<CrowdsecDecision>> getCrowdsecDecisions() {
+        return httpClient.get()
+            .uri(GET_DECISION)
+            .responseSingle((response, body) -> {
+                if (response.status().code() == 200) {
+                    return body.asString().map(this::parseCrowdsecDecisions);
+                } else {
+                    return Mono.error(new RuntimeException("Request failed with status code " + response.status().code()));
```

Comment:
Can we include the response body as well?

IMO we should strong type exception for CrowdSec e.g. `CrowSecException`. Then upon CrowdSec failure at the protocol layer, we can easily catch and handle the exception (rather than fail user requests).

e.g. An admin mistake setting up wrongly the CrowdSec apikey should not fail user SMTP requests.

Reply:
Instead of creating an exception, we can add logs to James for explaining why request to crowdsec failed (wrong key, wrong crowdsec url, ...)